cmd/gitops/add/profiles/cmd.go

```diff
@@ -0,0 +1,131 @@
+		`,
+		RunE: addProfileCmdRunE(client),
+	}
+
+	cmd.Flags().StringVar(&opts.Name, "name", "", "Name of the profile")
+	cmd.Flags().StringVar(&opts.Version, "version", "latest", "Version of the profile")
+	cmd.Flags().StringVar(&opts.ConfigRepo, "config-repo", "", "URL of external repository (if any) which will hold automation manifests")
+	cmd.Flags().StringVar(&opts.Cluster, "cluster", "", "Name of the cluster to add the profile to")
+	cmd.Flags().StringVar(&opts.Port, "port", server.DefaultPort, "Port the profiles API is running on")
+	cmd.Flags().BoolVar(&opts.AutoMerge, "auto-merge", false, "If set, 'gitops add profile' will merge automatically into the set --branch")
+
+	return cmd
+}
+
+func addProfileCmdRunE(client *resty.Client) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		log := internal.NewCLILogger(os.Stdout)
+		fluxClient := flux.New(osys.New(), &runner.CLIRunner{})
+		factory := services.NewFactory(fluxClient, log)
+		providerClient := internal.NewGitProviderClient(os.Stdout, os.LookupEnv, auth.NewAuthCLIHandler, log)
+
+		if err := validateAddOptions(opts); err != nil {
+			return err
+		}
+
+		ns, err := cmd.Parent().Parent().Flags().GetString("namespace")
+		if err != nil {
+			return err
+		}
+		opts.Namespace = ns
+
+		config, err := clientcmd.BuildConfigFromFlags("", filepath.Join(homedir.HomeDir(), ".kube", "config"))
+		if err != nil {
+			return fmt.Errorf("error initializing kubernetes config: %w", err)
+		}
+
+		clientSet, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			return fmt.Errorf("error initializing kubernetes client: %w", err)
+		}
+
+		kubeClient, _, err := kube.NewKubeHTTPClient()
+		if err != nil {
+			return fmt.Errorf("failed to create kube client: %w", err)
+		}
+
+		ctx := context.Background()
+		_, gitProvider, err := factory.GetGitClients(ctx, kubeClient, providerClient, services.GitConfigParams{
+			ConfigRepo:       opts.ConfigRepo,
+			Namespace:        opts.Namespace,
+			IsHelmRepository: true,
+			DryRun:           false,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get git clients: %w", err)
+		}
+
+		return profiles.NewService(clientSet, log).Add(ctx, gitProvider, opts)
+	}
+}
+
+func validateAddOptions(opts profiles.AddOptions) error {
+	if opts.Name == "" {
+		return errors.New("--name should be provided")
+	}
+
+	if models.ApplicationNameTooLong(opts.Name) {
+		return fmt.Errorf("--name value is too long: %s; must be <= %d characters",
+			opts.Name, models.MaxKubernetesResourceNameLength)
+	}
+
+	if strings.HasPrefix(opts.Name, "wego") {
+		return fmt.Errorf("the prefix 'wego' is used by weave gitops and is not allowed for a profile name")
+	}
+
+	if opts.ConfigRepo == "" {
+		return errors.New("--config-repo should be provided")
+	}
+
+	if opts.Cluster == "" {
+		return errors.New("--cluster should be provided")
+	}
```

Comment:
Required flags is a feature provided by Cobra, so you could do this in the command initialization code:

```golang
	requiredFlags := []string{"name", "config-repo", "cluster"}
	for _, f := range requiredFlags {
		if err := cobra.MarkFlagRequired(cmd.Flags(), f); err != nil {
			panic(fmt.Errorf("unexpected error: %w", err))
		}

	}
```

```suggestion
	if models.ApplicationNameTooLong(opts.Name) {
		return fmt.Errorf("--name value is too long: %s; must be <= %d characters",
			opts.Name, models.MaxKubernetesResourceNameLength)
	}

	if strings.HasPrefix(opts.Name, "wego") {
		return fmt.Errorf("the prefix 'wego' is used by weave gitops and is not allowed for a profile name")
	}
```

Reply:
Thank you for this suggestion, I added this in the latest commit. :)